### ISSUE TITLE
Build failing due to webpack misconfiguration error on 4.0.1

### ISSUE THREAD

**Issue by knowler** [roots/bud#56]
- [x] I've read the [guidelines for Contributing to Roots Projects](https://github.com/roots/.github/blob/master/CONTRIBUTING.md)
- [x] This request isn't a duplicate of an existing issue
- [x] I've read the [docs](https://roots.io/documentation/) and followed them (if applicable)
- [x] This is not a personal support request that should be posted on the [Roots Discourse](https://discourse.roots.io/) community

## Description

Using Bud 4.0.1 with Node 16.1.0 and Yarn 1.22.10 (on macOS 11.3.1) running `yarn bud build` fails with the following:

<details><summary>Output</summary>

This isn't the full output:

```
$ /Users/knowler/code/trying-bud/node_modules/.bin/bud build


 █▄▄ █ █ █▀▄
 █▄█ █▄█ █▄▀
















 ERROR  Invalid configuration object. Webpack has been initialized using a configuration object that does not match the API schema.
         - configuration.cache.type should be "memory".
           -> In memory caching.

 -  - configuration.cache.type should be "memory".
 -    -> In memory caching.
 - validate (node_modules/schema-utils/dist/validate.js:104:11)
 - validateSchema (node_modules/webpack/lib/validateSchema.js:79:2)
 - create (node_modules/webpack/lib/webpack.js:109:24)
 - webpack (node_modules/webpack/lib/webpack.js:147:32)
 - Object.f (node_modules/webpack/lib/index.js:54:15)
 - default_1.compile (node_modules/@roots/bud-compiler/lib/cjs/Compiler/index.js:38:42)
 -  (node_modules/@roots/bud-dashboard/lib/cjs/hooks/useCompilation.js:52:39)
 - commitHookEffectList (node_modules/react-reconciler/cjs/react-reconciler.development.js:12025:26)
 - commitPassiveHookEffects (node_modules/react-reconciler/cjs/react-reconciler.development.js:12059:11)
 - Object.invokeGuardedCallbackImpl (node_modules/react-reconciler/cjs/react-reconciler.development.js:11563:10)

/Users/knowler/code/trying-bud/node_modules/yoga-layout-prebuilt/yoga-layout/build/Release/nbind.js:53
        throw ex;
        ^

Error: Command failed with exit code 128: git rev-parse --short HEAD --no-color
fatal: Needed a single revision
    at makeError (/Users/knowler/code/trying-bud/node_modules/execa/lib/error.js:59:11)
    at handlePromise (/Users/knowler/code/trying-bud/node_modules/execa/index.js:114:26)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  shortMessage: 'Command failed with exit code 128: git rev-parse --short HEAD --no-color',
  command: 'git rev-parse --short HEAD --no-color',
  exitCode: 128,
  signal: undefined,
  signalDescription: undefined,
  stdout: '',
  stderr: 'fatal: Needed a single revision',
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
}
error Command failed with exit code 7.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

</details>

<details><summary><code>bud.config.js</code></summary>

```js
module.exports = bud => bud.entry('app', ['app.js']);
```

</details>

<details><summary><code>package.json</code></summary>

```json
{
  "private": true,
  "devDependencies": {
    "@roots/bud": "^4.0.1",
    "@roots/bud-cli": "^4.0.1"
  }
}
```

</details>

<details><summary>Project structure</summary>

```
.
├── src
│  └── app.js
├── bud.config.js
├── package.json
└── yarn.lock
```

</details>

**Comment by kellymears** [comment]
This is fixed in #59  and will be released in v4.1.0.

For now, you can get around this by setting `bud.persist(false)` or `bud.persist({type: 'memory'})`:

```ts
module.exports = bud => bud.entry('app', ['app.js']).persist(false);
```

Thanks for reporting this issue!

**Comment by kellymears** [comment]
In v4.1.0 the default caching strat will be `memory`. You will need to call `bud.persist()` (which no longer takes any arguments) to enable fs caching (I think it's worth it). 

You can also supply your own webpack cache config using the `build/cache` hook:

```ts
bud.hooks.on('build/cache', cache => ({
  ...cache,
  // overrides
}))
```
